samplers/samplers.go

```diff
@@ -28,98 +28,132 @@ type DDMetric struct {
 	Interval   int32         `json:"interval,omitempty"`
 }
 
-type tsvField int
+type Aggregate int
 
 const (
-	// the order in which these appear determines the
-	// order of the fields in the resultant TSV
-	tsvName tsvField = iota
-	tsvTags
-	tsvMetricType
-
-	// The hostname attached to the metric
-	tsvHostname
-
-	// The hostname of the server flushing the data
-	tsvVeneurHostname
-
-	tsvDeviceName
-	tsvInterval
-
-	tsvTimestamp
-	tsvValue
-
-	// This is the _partition field
-	// required by the Redshift IncrementalLoader.
-	// For our purposes, the current date is a good partition.
-	tsvPartition
+	AggregateMin Aggregate = 1 << iota
+	AggregateMax
+	AggregateMedian
+	AggregateAverage
+	AggregateCount
+	AggregateSum
 )
 
-var tsvSchema = [...]string{
-	tsvName:           "tsvName",
-	tsvTags:           "tsvTags",
-	tsvMetricType:     "tsvMetricType",
-	tsvHostname:       "tsvHostname",
-	tsvDeviceName:     "tsvDeviceName",
-	tsvInterval:       "tsvInterval",
-	tsvVeneurHostname: "tsvVeneurHostname",
-	tsvTimestamp:      "tsvTimestamp",
-	tsvValue:          "tsvValue",
-	tsvPartition:      "tsvPartition",
+var AggregatesLookup = map[string]Aggregate{
+	"min":    AggregateMin,
+	"max":    AggregateMax,
+	"median": AggregateMedian,
+	"avg":    AggregateAverage,
+	"count":  AggregateCount,
+	"sum":    AggregateSum,
 }
 
-// String returns the field name.
-// eg tsvName.String() returns "Name"
-func (f tsvField) String() string {
-	return fmt.Sprintf(strings.Replace(tsvSchema[f], "tsv", "", 1))
+type HistogramAggregates struct {
+	Value Aggregate
+	Count int
 }
 
-// each key in tsvMapping is guaranteed to have a unique value
-var tsvMapping = map[string]int{}
-
-func init() {
-	for i, field := range tsvSchema {
-		tsvMapping[field] = i
-	}
+var aggregates = [...]string{
+	AggregateMin:     "min",
+	AggregateMax:     "max",
+	AggregateMedian:  "median",
+	AggregateAverage: "avg",
+	AggregateCount:   "count",
+	AggregateSum:     "sum",
 }
 
-// encodeCSV generates a newline-terminated CSV row that describes
+// EncodeCSV generates a newline-terminated CSV row that describes
 // the data represented by the DDMetric.
 // The caller is responsible for setting w.Comma as the appropriate delimiter.
 // For performance, encodeCSV does not flush after every call; the caller is
 // expected to flush at the end of the operation cycle
-func (d DDMetric) encodeCSV(w *csv.Writer, partitionDate *time.Time, hostname string) error {
+func (d DDMetric) EncodeCSV(w *csv.Writer, partitionDate *time.Time, hostName string) error {
```

Comment:
did we agree that we would move this into s3 package? not saying we have to do it now but just making sure we remember to do it